### PR TITLE
Karpenter update

### PR DESCRIPTION
**Pull request by myaser**
in the latest version of karpenter >= v0.32, the CRDs have been renamed and their structure have slightly changed as the api graduated from alpha to beta
in addition, the couldformation templates security have been strengthened by adding limitations on the resources that can be manipulated by the controller

this PR adapts the manifests and cloudformation templates of the newer version

### CODE REVIEW

File: cluster/manifests/z-karpenter/webhooks.yaml [hunk 1]
```diff
@@ -1,128 +0,0 @@
-{{ if eq .Cluster.ConfigItems.karpenter_pools_enabled "true"}}
```
Comment (mikkeloscar) [inline cluster/manifests/z-karpenter/webhooks.yaml:1]: note: we should also make sure to delete these resources via `cluster/manifests/deletions.yaml` file. Currently they would not be deleted with `karpenter_pools_enabled=true`

Comment (mikkeloscar) [reply cluster/manifests/z-karpenter/webhooks.yaml:1]: Same for the CRDs "renaming" there we should also delete the old ones via `deletions.yaml`

Comment (myaser) [reply cluster/manifests/z-karpenter/webhooks.yaml:1]: fixed, 
the only thing that could not be done in this iteration is the CRDs.
they can not be deleted before removing their instances
so I kept them under the `karpenter_pools_enabled=true` for now until being cleaned up later after migration is complete

File: test/e2e/Dockerfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ RUN chmod +x /usr/bin/kubectl
 COPY --from=builder /go/bin/ginkgo /usr/local/bin/ginkgo
 
 # copy CLM
-COPY --from=container-registry.zalando.net/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
+COPY --from=container-registry-test.zalando.net/teapot/cluster-lifecycle-manager:pr-744-3 /clm /usr/bin/clm
```
Comment (myaser) [inline test/e2e/Dockerfile]:
```suggestion
COPY --from=container-registry-test.zalando.net/teapot/cluster-lifecycle-manager:pr-744-5 /clm /usr/bin/clm
```

File: test/e2e/Dockerfile [hunk 3]
```diff
@@ -26,7 +26,7 @@ RUN chmod +x /usr/bin/kubectl
 COPY --from=builder /go/bin/ginkgo /usr/local/bin/ginkgo
 
 # copy CLM
-COPY --from=container-registry.zalando.net/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
+COPY --from=container-registry-test.zalando.net/teapot/cluster-lifecycle-manager:pr-744-5 /clm /usr/bin/clm
```
Comment (myaser) [inline test/e2e/Dockerfile]:
```suggestion
COPY --from=container-registry.zalando.net/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
```

File: cluster/manifests/deletions.yaml [hunk 4]
```diff
@@ -184,48 +220,31 @@ post_apply:
   kind: ClusterRole
 - name: karpenter-core
   kind: ClusterRoleBinding
-- name: karpenter-webhook
+- name: karpenter
   kind: ClusterRole
-- name: karpenter-webhook
+- name : karpenter
```
Comment (mikkeloscar) [inline cluster/manifests/deletions.yaml]: looks wrong with a space here :)

File: cluster/manifests/deletions.yaml [hunk 5]
```diff
@@ -164,14 +164,50 @@ post_apply:
...
+  kind: Secret
+  namespace: kube-system
+- name: defaulting.webhook.karpenter.k8s.aws
+  kind: MutatingWebhookConfiguration
+- name: validation.webhook.karpenter.k8s.aws
+  kind: ValidatingWebhookConfiguration
+- name: validation.webhook.config.karpenter.sh
+  kind: ValidatingWebhookConfiguration
+- name: validation.webhook.karpenter.sh
+  kind: ValidatingWebhookConfiguration
```
Comment (mikkeloscar) [inline cluster/manifests/deletions.yaml:197]: Are all these legacy things that should be deleted for the update?

Maybe add a comment to make it clear? :)

**Comment by mikkeloscar** [comment]
:+1:

**Comment by myaser** [comment]
👍🏻

**Pull request merged**